firmware/mcal/stm32f767/periph/pwm.h

```diff
@@ -60,10 +60,22 @@ class PWMOutput : public shared::periph::PWMOutput {
         return float(pulse) / float(period) * 100.0f;
     }
 
+    void SetFrequency(float frequency) override {
+        frequency_ = std::max((float) 0, frequency);
```

Comment:
instead of casting `0` to float, just use `0.0f`

Reply:
In fact, clamping to `0` will cause a divide by zero error, so our min value should be something like 0.5 Hz. Can you find the stm's lowest possible PWM frequency?

Reply:
rather than remapping to 0 to any value, I think we should make it clear an error has occured. Since we do not have good error support, we should maybe do nothing for now or turn the pwm off completely.

Reply:
This is the plan once errors are ready. If I set the freq too low, I would expect the PWM to still run but at a different freq. Turning it off completely would be harder to debug than operating at the wrong frequency